packages/metrics-util/test/nodejs/NodeJsProcessor_test.js

```diff
@@ -7,54 +7,62 @@
 
 const { expect } = require('chai');
 const semver = require('semver');
-
-const { metrics: coreMetrics } = require('@instana/core');
-
+const sinon = require('sinon');
 const { delay, retry } = require('../../../core/test/test_util');
-const config = require('@instana/core/test/config');
-
-const sharedMetrics = require('@instana/shared-metrics');
-
+const testConfig = require('@instana/core/test/config');
 const NodeJsProcessor = require('../../src/nodejs/NodeJsProcessor');
 
 describe('Node.js processor', function () {
-  this.timeout(config.getTestTimeout());
+  this.timeout(testConfig.getTestTimeout());
+  const coreAndSharedMetricsDataSource = sinon.stub();
+  coreAndSharedMetricsDataSource.on = sinon.stub();
+  coreAndSharedMetricsDataSource.activate = sinon.stub();
+  coreAndSharedMetricsDataSource.deactivate = sinon.stub();
+  coreAndSharedMetricsDataSource.hasRefreshedAtLeastOnce = sinon.stub();
+  coreAndSharedMetricsDataSource.getRawData = sinon.stub();
 
   let dataProcessor;
 
   before(() => {
-    coreMetrics.registerAdditionalMetrics(sharedMetrics.allMetrics);
-    coreMetrics.init({
-      metrics: {
-        transmissionDelay: 1000
-      }
-    });
-
-    dataProcessor = new NodeJsProcessor(coreMetrics, 42);
+    dataProcessor = new NodeJsProcessor(coreAndSharedMetricsDataSource, 42);
```

Comment:
This is the right way to test the unit test.
We pass in a mock.

In the old code we had to init the coreMetrics to run a unit test for "NodeJsProcessor".
"NodeJsProcessor" does not care about the source and should not depend on the initialization of it.